### ISSUE TITLE
Virtual Panic Button as a homekit button accessory

### ISSUE THREAD

**Issue by sapireli** [node-alarm-dot-com/homebridge-node-alarm-dot-com#124]
Alarm.com provides a virtual panic buttons in the mobile app to trigger a panic i.e. burglary silent alarm etc. It would be great to expose such buttons as homebridge accessories... Ideally once any specific panic button or switch is triggered it stays on until the alarm is cleared etc. 

![image](https://github.com/node-alarm-dot-com/homebridge-node-alarm-dot-com/assets/114778791/b18aff67-4c06-40d6-ad8e-5f4e3375ff42)

More info on the virtual panic buttons here: 
https://answers.alarm.com/Customer/Website_and_App/System_Management/System_Usage/Trigger_a_panic_using_the_app

**Comment by sapireli** [comment]
I agree with all your points. I did have one thought about the implementation. We could implement it as a "door lock" accessory which requires face-id or device unlocking to open so you couldn't really trigger with Siri. 

However, triggering PANIC with Siri could also be a feature not a bug. i.e. i fell down call an ambulance...

**Comment by sapireli** [comment]
Also, I learned today that home [homebridge-ring](https://github.com/dgreif/ring/tree/main/packages/homebridge-ring#:~:text=test%20these%20on.-,Panic%20Buttons,-These%20can%20be) actually implements panic buttons in homekit as follows (from their readme):

`Panic Buttons
These can be added by setting showPanicButtons: true in your config
Creates Burglar Alarm and Fire Alarm switches in a new Panic Buttons device in HomeKit
Use these at your own risk. I do not guarantee functionality in case of emergency, nor do I take responsibility for any false alarms
If either switch is turned on, you will receive a call from Ring monitoring to verify the emergency, and then authorities will be dispatched`

**Comment by chase9** [comment]
I'm still pretty uneasy about it, but locking the functionality behind an "advanced" setting does make me feel a bit better.

**Comment by sapireli** [comment]
how would i go about this? is there an official api or is the whole thing reverse engineered using a mitm proxy?

**Comment by ngori** [comment]
@sapireli There is no official API support.  As far as I know, all functionality to date has been reversed engineered.  I just captured the API calls for garage doors when I added support there.  I don't have a panic button on my setup so I'm not sure how those work with respect to the apps and web interfaces
